File: Python/Machine Learning Starter Templates/Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression.R
#Read dataset from csv file
dataset = read.csv('Salary_Data.csv')
# dataset = dataset[, 2:3]

#Divide dataset into test set and training set
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#Feature Scaling
# training_set[,2:3] = scale(training_set[,2:3])
# test_set[,2:3] = scale(test_set[,2:3])

regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

y_pred = predict(regressor, newdata = test_set)

library(ggplot2)
ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') + 
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, new_data = training_set)),
            color = 'blue') +
  ggtitle('Salary vs Experience (Training Set)') + 
  xlab("Years of experience") + 
  ylab("Salary")

ggplot() + 
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') + 
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, new_data = training_set)),
            color = 'blue') +
  ggtitle('Salary vs Experience (Test Set)') + 
  xlab("Years of experience") + 
  ylab("Salary")
  
  

File: Python/Machine Learning Starter Templates/Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression.py
#Simple Linear Regression
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Read dataset from csv file
dataset = pd.read_csv('Salary_Data.csv')

#Store the first three columns into X
X = dataset.iloc[:,:-1].values

#Store the last column into Y
Y = dataset.iloc[:,1].values 


#Divide dataset into test set and training set
from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X,Y, test_size = 1/3, random_state = 0)

#Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""

#Fitting Simple Linear Regression to the Training Set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, Y_train)

#Predicting the Test set results
Y_pred = regressor.predict(X_test)

#Visualizing the Training Set results
plt.scatter(X_train, Y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title("Salary vs Experience (Training Set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()

#Visualizing the Test Set results
plt.scatter(X_test, Y_test, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title("Salary vs Experience (Test Set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()